src/js/modules/infragistics.ui.editors.js

```diff
@@ -10396,15 +10411,17 @@
 			}
 		},
 		getSelectedDate: function() {
-			/* Gets selected date.
+			/* Gets selected date as a date object. This method can be used when dataMode is set as either displayModeText or editModeText.
+			In such cases the value() method will not return date object and getSelectedDate() can be used to replace that functionality.
 			```
 			$(".selector").%%WidgetName%%("getSelectedDate");
 			```
 				returnType="date" */
 			return this._dateObjectValue;
```

Comment:
That'd be the internal date object the editor is based around, if it's wrong at any given point something must've gone terribly wrong elsewhere. Could re-create the object (from the internal one) purely to avoid unintentional changes to the return value from applying to the editor. There's `selectDate` for that after all :)